clients/src/main/java/org/apache/kafka/common/Cluster.java

```diff
@@ -42,17 +43,32 @@
      * Create a new cluster with the given nodes and partitions
      * @param nodes The nodes in the cluster
      * @param partitions Information about a subset of the topic-partitions this cluster hosts
+     * @deprecated Use the Cluster constructor with 4 parameters
      */
+    @Deprecated
```

Comment:
Fair enough. Even using it in a test case would be kind of weird since it's not reachable from the producer or consumer APIs directly. I guess we should locate stuff like this in `common.internals` from now on.